drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-marshaller/src/main/resources/org/drools/workbench/scenariosimulation/kogito/marshaller/jre/javax/xml/namespace/QName.java

```diff
@@ -0,0 +1,169 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package javax.xml.namespace;
+
+import java.util.Objects;
+
+import javax.xml.XMLConstants;
+
+import jsinterop.annotations.JsConstructor;
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsProperty;
+import jsinterop.annotations.JsType;
+
+/**
+ * GWT Super-source version of javax.xml.namespace.QName.
+ */
+@JsType(namespace = JsPackage.GLOBAL)
+public class QName {
+
+    @JsProperty(name = "namespaceURI")
+    public final native String getNamespaceURI();
+
+    @JsProperty(name = "namespaceURI")
+    public final native void setNamespaceURI(final String namespaceURI);
+
+    @JsProperty(name = "localPart")
+    public final native String getLocalPart();
+
+    @JsProperty(name = "localPart")
+    public final native void setLocalPart(final String localPart);
+
+    @JsProperty(name = "prefix")
+    public final native String getPrefix();
+
+    @JsProperty(name = "prefix")
+    public final native void setPrefix(final String prefix);
+
+    @JsProperty(name = "key")
+    public final native String getKey();
+
+    @JsProperty(name = "key")
+    public final native void setKey(String key);
+
+    @JsProperty(name = "string")
+    public final native String getString();
+
+    @JsProperty(name = "string")
+    public final native void setString(String string);
+
+    @JsConstructor
+    public QName(final String namespaceURI,
+                 final String localPart,
+                 final String prefix) {
+        if (Objects.isNull(namespaceURI)) {
+            setNamespaceURI(XMLConstants.NULL_NS_URI);
+        } else {
+            setNamespaceURI(namespaceURI);
+        }
+
+        if (Objects.isNull(localPart)) {
+            throw new IllegalArgumentException("local part cannot be \"null\" when creating a QName");
+        }
+        setLocalPart(localPart);
+
+        if (Objects.isNull(prefix)) {
+            throw new IllegalArgumentException("prefix cannot be \"null\" when creating a QName");
+        }
+        setPrefix(prefix);
+
+        //jsonix JSON properties
+        setKey(toString());
+        String retrievedPrefix = getPrefix();
+        final String usedPrefix = (retrievedPrefix == null || retrievedPrefix.isEmpty()) ?  "" : retrievedPrefix + ":";
+        final String string = "{" + getNamespaceURI() + "}" + usedPrefix + getLocalPart();
+        setString(string);
+    }
+
+    @Override
+    public String toString() {
+        if (getNamespaceURI().equals(XMLConstants.NULL_NS_URI)) {
```

Comment:
Could we please use `Objects.equals` also here?

Reply:
@jomarko I don't see problems do to that, done!